protoc-gen-swagger/main.go

```diff
@@ -0,0 +1,115 @@
+package main
```

Comment:
I think you're referring to https://github.com/golang/protobuf/tree/master/protoc-gen-go which is the only protoc-gen-go and grpc-plugin I'm aware of. If I'm talking past you, please disregard.

It sounds to me like you're trying to ensure that the two generators are going to be versioned the same. Respectfully, I think this concern is not something that can be solved at this level. If you're a repository maintainer, you need to be tracking all the code generators you depend upon and staying up to date. I've several times gotten broken by drift between protoc-gen-go and grpc-gateway's gw definitions (not to mention the other generators). I think that keeping things in sync is a known issue with the entire ecosystem.

This is especially true because the code generation doesn't stop here. With the introduction of swagger support I am hooking up a JS code generator based on swagger-codegen (I will push that upstream). There are now 4 independent code generators in my project (protoc, protoc-gen-go, protoc-gen-grpg-gateway, and swagger-codegen). I expect to become familiar with the upgrade process for each.

Further, using the protoc-gen-go has a fairly obtuse parameter passing syntax --gen-go_out=plugins=grpc:. is puzzling at first, to say the least. I now know the syntax but to learn it I had to put log.Printf statements into the code, recompile it and waste a day or two on digging through code to learn why gRPC bindings weren't being created. Having these multiple modes which each produce valid outputs may be "easier" for you/me, but I think that it raises the barrier to entry for code generation which will reduce adoption. In summary, I feel the principle of UNIX is in effect here, "do one thing and do it well."

If your concern is keeping the code behind each generator in sync, I think that is orthogonal to the issue of having multiple binaries. We could have one folder that produces two binaries or two folders that produced a single one, it is just a matter of configuration. What code do you have in mind as being difficult to keep in sync between the two? Maybe we can find a way to abstract that logic into something a bit more pleasant to work with.

Looking forward to your responses and thanks for reviewing my code.